dump.go

```diff
@@ -63,11 +64,22 @@ func DumpRequest(req *http.Request) ([]byte, error) {
 		req.ProtoMinor,
 	)
 
-	for key := range req.Header {
+	sortedKeys := sortedHeaderKeys(req)
+	for _, key := range sortedKeys {
 		val := req.Header.Get(key)
 		fmt.Fprintf(buf, "%s: %s\n", withColor(31, key), withColor(32, val))
 	}
 
 	err := writeBody(buf, req)
 	return buf.Bytes(), err
 }
+
+func sortedHeaderKeys(req *http.Request) []string {
+	keys := make([]string, len(req.Header))
+	for header := range req.Header {
+		keys = append(keys, header)
```

Comment:
this might not gonna work as you expect :(
1 you're creating a slice of size (req.Header)
2 to that already allocated slice, you're appending the keys
The results is and array of size len(req.Header) * 2. where the first half is empty.